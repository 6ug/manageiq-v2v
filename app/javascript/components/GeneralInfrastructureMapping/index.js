import React from 'react';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import { Form, Grid } from 'patternfly-react';

class GeneralInfrastructureMappingContainer extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <form className="form-horizontal">
        <Field
          name="name"
          label={__('Name')}
          component={renderField}
          type="text"
        />
        <Field
          name="description"
          label={__('Description')}
          component={renderField}
          type="textarea"
        />
      </form>
    );
  }
}

const renderField = ({
  input,
  label,
  controlId,
  type,
  meta: { pristine, touched, error },
  ...props
}) => {
  const formGroupProps = { key: { label }, controlId, ...props };

  return (
    <Form.FormGroup {...formGroupProps}>
      <Grid.Col componentClass={Form.ControlLabel} sm={2}>
        {label}
      </Grid.Col>
      <Grid.Col sm={9}>
        <Form.FormControl
          {...input}
          type={type}
          componentClass={type === 'text' ? undefined : type}
        />
      </Grid.Col>
    </Form.FormGroup>
  );
};

// Decorate with reduxForm(). It will read the initialValues prop provided by connect()
const GeneralInfrastructureMappingForm = reduxForm({
  form: 'generalInfrastructureMapping', // a unique identifier for this form
  destroyOnUnmount: false // preserve form data
  // forceUnregisterOnUnmount: true, // unregister fields on unmount
})(GeneralInfrastructureMappingContainer);

export default connect()(GeneralInfrastructureMappingForm);
