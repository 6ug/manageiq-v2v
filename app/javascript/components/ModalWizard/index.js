import ModalWizard from './ModalWizard';
import ModalWizardBody from './ModalWizardBody';
import ModalWizardStateProvider from './ModalWizardStateProvider';

// TODO maybe move these up to patternfly-react?
ModalWizard.StateProvider = ModalWizardStateProvider;
ModalWizard.Body = ModalWizardBody;

export default ModalWizard;
