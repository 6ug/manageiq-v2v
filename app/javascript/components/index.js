import MappingWizardContainer from './MappingWizard';
import PlanWizardContainer from './PlanWizard';
import Dashboard from '../migration/scenes/Dashboard';
import Overview from '../migration/scenes/Overview';
import IsoDate from './dates/IsoDate';
import LongDateTime from './dates/LongDateTime';
import RelativeDateTime from './dates/RelativeDateTime';
import ShortDateTime from './dates/ShortDateTime';
import MiqV2vUi from '../migration/containers/Application';

export const coreComponents = [
  {
    name: 'MappingWizardContainer',
    type: MappingWizardContainer,
    data: { url: '/api/sourceClusters' },
    store: true
  },
  {
    name: 'PlanWizardContainer',
    type: PlanWizardContainer,
    data: { url: '/api/migrationPlans' },
    store: true
  },
  {
    name: 'Dashboard',
    type: Dashboard,
    data: {}
  },
  {
    name: 'Overview',
    type: Overview,
    data: {},
    store: true
  },
  {
    name: 'RelativeDateTime',
    type: RelativeDateTime,
    data: true,
    store: false
  },
  {
    name: 'LongDateTime',
    type: LongDateTime,
    data: true,
    store: false
  },
  {
    name: 'ShortDateTime',
    type: ShortDateTime,
    data: true,
    store: false
  },
  {
    name: 'IsoDate',
    type: IsoDate,
    data: true,
    store: false
  },
  { name: 'v2v_ui_plugin', type: MiqV2vUi }
];

export const componentSettings = component =>
  coreComponents.find(n => n.type === component);
