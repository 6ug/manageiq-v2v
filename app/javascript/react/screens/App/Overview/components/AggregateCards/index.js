import PendingTransformationPlans from './PendingTransformationPlans/PendingTransformationPlans';
import MigrationsInProgress from './MigrationsInProgress/MigrationsInProgress';
import MigrationsComplete from './MigrationsComplete/MigrationsComplete';
import InfrastructureMappings from './InfrastructureMappings/InfrastructureMappings';

export {
  PendingTransformationPlans,
  MigrationsInProgress,
  MigrationsComplete,
  InfrastructureMappings
};
