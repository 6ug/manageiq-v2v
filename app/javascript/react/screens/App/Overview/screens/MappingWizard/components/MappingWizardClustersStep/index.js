import { connect } from 'react-redux';
import MappingWizardClustersStep from './MappingWizardClustersStep';
import * as MappingWizardClustersStepActions from './MappingWizardClustersStepActions';

import reducer from './MappingWizardClustersStepReducer';

export const reducers = { mappingWizardClustersStep: reducer };

const mapStateToProps = ({ mappingWizardClustersStep, form }, ownProps) => ({
  ...mappingWizardClustersStep,
  ...ownProps.data,
  targetProvider: form.mappingWizardGeneralStep.values.targetProvider
});

const mergeProps = (stateProps, dispatchProps, ownProps) => Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(
  mapStateToProps,
  MappingWizardClustersStepActions,
  mergeProps
)(MappingWizardClustersStep);
