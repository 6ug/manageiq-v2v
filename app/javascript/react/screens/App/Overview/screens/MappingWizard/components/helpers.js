import { V2V_TARGET_PROVIDERS } from '../MappingWizardConstants';

export const getClusterOptions = clusterMappings => {
  const sourceClustersWithAssociatedTargetClusters = clusterMappings.reduce(
    (mappings, targetClusterWithSourceClusters) => {
      const { nodes: sourceClusters, ...targetCluster } = targetClusterWithSourceClusters;
      const sourceToTargetMappings = sourceClusters.map(sourceCluster => ({
        sourceCluster,
        targetCluster,
        sourceClusterMappedToTargetCluster: {
          name: `${sourceCluster.name} (${targetCluster.name})`,
          id: sourceCluster.id
        }
      }));
      return mappings.concat(sourceToTargetMappings);
    },
    []
  );

  return sourceClustersWithAssociatedTargetClusters.map(
    ({ sourceClusterMappedToTargetCluster }) => sourceClusterMappedToTargetCluster
  );
};

export const multiProviderTargetLabel = (providerType, wizardStep) => {
  switch (providerType) {
    case V2V_TARGET_PROVIDERS[1].id:
      switch (wizardStep) {
        case 'cluster':
          return __('Target Provider \\ Project');
        case 'network':
          return __('Target Project \\ Network');
        default:
          return null;
      }
    default:
      switch (wizardStep) {
        case 'cluster':
          return __('Target Provider \\ Datacenter \\ Cluster');
        case 'network':
          return __('Target Networks');
        default:
          return null;
      }
  }
};
