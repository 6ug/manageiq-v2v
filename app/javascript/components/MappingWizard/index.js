import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import { noop, selectKeys } from 'patternfly-react';
import * as MappingWizardActions from '../../redux/actions/mappingWizard';
import ModalWizard from '../ModalWizard';
import MappingWizardBody from './MappingWizardBody';

class MappingWizardContainer extends React.Component {
  componentDidMount() {
    const { url, fetchSourceClusters } = this.props;

    fetchSourceClusters(url);
  }

  render() {
    const { sourceClusters, isFetching } = this.props;
    const modalProps = selectKeys(this.props, [
      'showWizard',
      'onHide',
      'onExited'
    ]);

    console.log('Source Clusters:', sourceClusters);

    return (
      <ModalWizard.StateProvider numSteps={5}>
        <ModalWizard
          {...modalProps}
          title={<FormattedMessage id="mappingWizard.title" />}
        >
          <MappingWizardBody loaded={!isFetching} />
        </ModalWizard>
      </ModalWizard.StateProvider>
    );
  }
}
MappingWizardContainer.propTypes = {
  onHide: PropTypes.func,
  onExited: PropTypes.func,
  showWizard: PropTypes.bool,
  url: PropTypes.string,
  fetchSourceClusters: PropTypes.func,
  sourceClusters: PropTypes.arrayOf(PropTypes.object),
  isFetching: PropTypes.bool
};
MappingWizardContainer.defaultProps = {
  url: '',
  fetchSourceClusters: noop,
  sourceClusters: [],
  isFetching: true
};

const mapStateToProps = (state, ownProps) => ({
  ...state.mappingWizard,
  ...ownProps.data
});

const mergeProps = (stateProps, dispatchProps, ownProps) =>
  Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(mapStateToProps, MappingWizardActions, mergeProps)(
  MappingWizardContainer
);
