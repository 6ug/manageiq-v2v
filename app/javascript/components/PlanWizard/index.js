import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import { noop, selectKeys } from 'patternfly-react';
import ModalWizard from '../ModalWizard';
import PlanWizardBody from './PlanWizardBody';

class PlanWizardContainer extends React.Component {
  constructor() {
    super();
    this.state = { loaded: false };
  }
  componentDidMount() {
    const that = this;
    setTimeout(() => {
      that.setState({ loaded: true }); // TODO replace me with a real API request
    }, 1000);
  }

  render() {
    console.log('planwizprops!!!!', this.props);
    const { loaded } = this.state;
    const modalProps = selectKeys(this.props, [
      'showWizard',
      'onHide',
      'onExited'
    ]);
    return (
      <ModalWizard.StateProvider numSteps={3}>
        <ModalWizard
          {...modalProps}
          title={<FormattedMessage id="planWizard.title" />}
        >
          <PlanWizardBody loaded={loaded} />
        </ModalWizard>
      </ModalWizard.StateProvider>
    );
  }
}

PlanWizardContainer.propTypes = {
  onHide: PropTypes.func,
  onExited: PropTypes.func,
  showWizard: PropTypes.bool
};
PlanWizardContainer.defaultProps = {
  onHide: noop,
  onExited: noop,
  showWizard: false
};

const mapStateToProps = (state, ownProps) => ({
  ...state.planWizard,
  ...ownProps.data
});

const mergeProps = (stateProps, dispatchProps, ownProps) =>
  Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(mapStateToProps, {}, mergeProps)(PlanWizardContainer);
