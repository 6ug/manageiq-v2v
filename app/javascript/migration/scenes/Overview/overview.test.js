import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { coreComponents } from '../../../components';
import componentRegistry from '../../../components/componentRegistry';

jest.mock('../../../components/componentRegistry');

componentRegistry.registerMultiple(coreComponents);

describe('Overview component', () => {
  it('renders the overview', () => {
    const Overview = componentRegistry.markup('Overview');
    const wrapper = mount(Overview);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
