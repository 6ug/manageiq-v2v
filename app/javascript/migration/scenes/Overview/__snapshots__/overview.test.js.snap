// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview component renders the overview 1`] = `
<I18nProviderWrapper(Overview)
  data={Object {}}
>
  <IntlProvider
    initialNow={2017-10-13T11:54:55.000Z}
    locale="en"
    messages={
      Object {
        "mappingWizard.back": "Back",
        "mappingWizard.cancel": "Cancel",
        "mappingWizard.next": "Next",
        "mappingWizard.title": "Infrastructure Mapping Wizard",
        "overview.emptyMappingCardButton": "Create Infrastructure Mapping",
        "overview.emptyMappingCardInfo": "Create mapping to later be used by a migration plan.",
        "overview.emptyMappingCardTitle": "Infrastructure Mappings",
        "overview.emptyPlanCardButton": "Create Migration Plan",
        "overview.emptyPlanCardInfo": "Create a migration plan to start migrating.",
        "overview.emptyPlanCardTitle": "Migration Plans",
      }
    }
  >
    <Overview
      data={Object {}}
    >
      <div
        className="row cards-pf"
        style={
          Object {
            "height": "100%",
            "overflow": "auto",
            "paddingBottom": 100,
          }
        }
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={12}
        >
          <div
            className="col-md-12"
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Equalizer
                  byRow={true}
                  enabled={[Function]}
                  nodes={[Function]}
                  property="height"
                >
                  <div
                    onLoad={[Function]}
                  >
                    <Col
                      bsClass="col"
                      componentClass="div"
                      xs={6}
                    >
                      <div
                        className="col-xs-6"
                      >
                        <div
                          className="spacer"
                        />
                        <div
                          className="card-pf"
                        >
                          <div
                            className="card-pf-body"
                          >
                            <div
                              className="blank-slate-pf"
                            >
                              <EmptyStateIcon
                                name="add-circle-o"
                                type="pf"
                              >
                                <div
                                  className="blank-slate-pf-icon"
                                >
                                  <Icon
                                    name="add-circle-o"
                                    type="pf"
                                  >
                                    <PatternflyIcon
                                      name="add-circle-o"
                                    >
                                      <span
                                        aria-hidden="true"
                                        className="pficon pficon-add-circle-o"
                                      />
                                    </PatternflyIcon>
                                  </Icon>
                                </div>
                              </EmptyStateIcon>
                              <EmptyStateTitle>
                                <h4
                                  className="h1 blank-slate-pf-title"
                                >
                                  <FormattedMessage
                                    id="overview.emptyMappingCardTitle"
                                    values={Object {}}
                                  >
                                    <span>
                                      Infrastructure Mappings
                                    </span>
                                  </FormattedMessage>
                                </h4>
                              </EmptyStateTitle>
                              <EmptyStateInfo>
                                <p
                                  className="blank-slate-pf-info"
                                >
                                  <FormattedMessage
                                    id="overview.emptyMappingCardInfo"
                                    values={Object {}}
                                  >
                                    <span>
                                      Create mapping to later be used by a migration plan.
                                    </span>
                                  </FormattedMessage>
                                </p>
                              </EmptyStateInfo>
                              <EmptyStateAction>
                                <div
                                  className="blank-slate-pf-main-action"
                                >
                                  <Button
                                    bsSize="large"
                                    bsStyle="primary"
                                    onClick={[Function]}
                                  >
                                    <Button
                                      active={false}
                                      block={false}
                                      bsClass="btn"
                                      bsSize="large"
                                      bsStyle="primary"
                                      disabled={false}
                                      onClick={[Function]}
                                    >
                                      <button
                                        className="btn btn-lg btn-primary"
                                        disabled={false}
                                        onClick={[Function]}
                                        type="button"
                                      >
                                        <FormattedMessage
                                          id="overview.emptyMappingCardButton"
                                          values={Object {}}
                                        >
                                          <span>
                                            Create Infrastructure Mapping
                                          </span>
                                        </FormattedMessage>
                                      </button>
                                    </Button>
                                  </Button>
                                </div>
                              </EmptyStateAction>
                            </div>
                          </div>
                        </div>
                      </div>
                    </Col>
                    <Col
                      bsClass="col"
                      componentClass="div"
                      xs={6}
                    >
                      <div
                        className="col-xs-6"
                      >
                        <div
                          className="spacer"
                        />
                        <div
                          className="card-pf"
                        >
                          <div
                            className="card-pf-body"
                          >
                            <div
                              className="blank-slate-pf"
                            >
                              <EmptyStateIcon
                                name="add-circle-o"
                                type="pf"
                              >
                                <div
                                  className="blank-slate-pf-icon"
                                >
                                  <Icon
                                    name="add-circle-o"
                                    type="pf"
                                  >
                                    <PatternflyIcon
                                      name="add-circle-o"
                                    >
                                      <span
                                        aria-hidden="true"
                                        className="pficon pficon-add-circle-o"
                                      />
                                    </PatternflyIcon>
                                  </Icon>
                                </div>
                              </EmptyStateIcon>
                              <EmptyStateTitle>
                                <h4
                                  className="h1 blank-slate-pf-title"
                                >
                                  <FormattedMessage
                                    id="overview.emptyPlanCardTitle"
                                    values={Object {}}
                                  >
                                    <span>
                                      Migration Plans
                                    </span>
                                  </FormattedMessage>
                                </h4>
                              </EmptyStateTitle>
                              <EmptyStateInfo>
                                <p
                                  className="blank-slate-pf-info"
                                >
                                  <FormattedMessage
                                    id="overview.emptyPlanCardInfo"
                                    values={Object {}}
                                  >
                                    <span>
                                      Create a migration plan to start migrating.
                                    </span>
                                  </FormattedMessage>
                                </p>
                              </EmptyStateInfo>
                              <EmptyStateAction>
                                <div
                                  className="blank-slate-pf-main-action"
                                >
                                  <Button
                                    bsSize="large"
                                    bsStyle="primary"
                                    onClick={[Function]}
                                  >
                                    <Button
                                      active={false}
                                      block={false}
                                      bsClass="btn"
                                      bsSize="large"
                                      bsStyle="primary"
                                      disabled={false}
                                      onClick={[Function]}
                                    >
                                      <button
                                        className="btn btn-lg btn-primary"
                                        disabled={false}
                                        onClick={[Function]}
                                        type="button"
                                      >
                                        <FormattedMessage
                                          id="overview.emptyPlanCardButton"
                                          values={Object {}}
                                        >
                                          <span>
                                            Create Migration Plan
                                          </span>
                                        </FormattedMessage>
                                      </button>
                                    </Button>
                                  </Button>
                                </div>
                              </EmptyStateAction>
                            </div>
                          </div>
                        </div>
                      </div>
                    </Col>
                  </div>
                </Equalizer>
              </div>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                  xs={12}
                >
                  <div
                    className="col-xs-12"
                  >
                    <div
                      className="card-pf"
                    >
                      <div
                        className="card-pf-heading"
                      >
                        <h2>
                          0 Migrations in Progress
                        </h2>
                      </div>
                      <div
                        className="card-pf-body"
                      >
                        <p>
                          content
                        </p>
                      </div>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                  xs={12}
                >
                  <div
                    className="col-xs-12"
                  >
                    <div
                      className="card-pf"
                    >
                      <div
                        className="card-pf-heading"
                      >
                        <h2>
                          0 Completed Migrations
                        </h2>
                      </div>
                      <div
                        className="card-pf-body"
                      >
                        <p>
                          content
                        </p>
                      </div>
                    </div>
                    <div
                      className="spacer"
                    />
                  </div>
                </Col>
              </div>
            </Row>
          </div>
        </Col>
      </div>
    </Overview>
  </IntlProvider>
</I18nProviderWrapper(Overview)>
`;
